src/core/GameKeyHandler.ttslua

```diff
@@ -57,6 +58,80 @@ function addDoomToAgenda()
   doomCounter.call("addVal", 1)
 end
 
+-- move the hovered object to the nearest empty slot on the playermat
+function takeCardIntoThreatArea(playerColor, hoveredObject)
+  -- only continue if an unlocked card
+  if hoveredObject == nil
+      or hoveredObject.type ~= "Card"
+      or hoveredObject.hasTag("Location")
+      or hoveredObject.locked then
+    broadcastToColor("Hover a non-location and try again.", playerColor, "Yellow")
+    return
+  end
+
+  local matColor = playmatApi.getMatColor(playerColor)
+  local mat = guidReferenceApi.getObjectByOwnerAndType(matColor, "Playermat")
+  -- do not continue if the threat area is already full
+  if playmatApi.getEncounterCardDrawPosition(matColor, false) == playmatApi.getEncounterCardDrawPosition(matColor, true) then 
+    broadcastToColor("Threat area is full.", playerColor,"Yellow")
+    return 
+  end
+  local pos = playmatApi.getEncounterCardDrawPosition(matColor, false)
+  
+  -- initialize list of objects to move
+  local moveTheseObjects = {}
+  
+  for _, obj in ipairs(searchLib.onObject(hoveredObject, "isTileOrToken")) do
+    table.insert(moveTheseObjects, obj)
+  end
+  
+  -- find out if the original card is on the green or red playmats
+  local result
+  local result = playmatApi.searchAroundPlaymat("Green", "isCard")
+  local originalMatColor
+  
+  for _, card in ipairs(result) do
+    if hoveredObject.getGUID() == card.getGUID() then
+      originalMatColor = "Green"
+    end
+  end
+  
+  local result = playmatApi.searchAroundPlaymat("Red", "isCard")
+  
+  for _, card in ipairs(result) do
+    if hoveredObject.getGUID() == card.getGUID() then
+      originalMatColor = "Red"
+    end
+  end
+  -- determine modifiers for the playmats
+  local modifierY
+  if originalMatColor == "Red" then
+    modifierY = 90
+  elseif originalMatColor == "Green" then
+    modifierY = -90
+  else
+    modifierY = 0
+  end
```

Comment:
Okay this needs to be more dynamic 😅

Reply:
Suggestions?

Reply:
Made something for you: https://github.com/argonui/SCED/pull/682